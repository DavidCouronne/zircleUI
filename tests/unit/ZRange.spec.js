import { shallowMount, createLocalVue } from '@vue/test-utils'
import zrange from '@/components/z-range.vue'
import zrangebar from '@/components/z-range-bar.vue'
import zircle from '@/index'
const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zrange, {
  propsData: {
    progress: 80,
    color: 'success',
    imagesrc: './images/test.png',
    label: 'Test label prop'
  },
  provide: function () {
    return {
      view: 'test'
    }
  },
  computed: {
    position: function () {
      return {X: 0, Y: 0, scale: 1, Xi: 0, Yi: 0, scalei: 1}
    }
  },
  slots: {
    default: '<p> lorem text </p>',
    media: '<div>youtube video</div>',
    imagesrc: './images/dummy.png',
    label: 'Test label slot',
    extension: '<div></div>'
  },
  stubs: {
    'z-range-bar': zrangebar
  },
  localVue
})
describe('z-range.vue', () => {
  it('Renders props.range by default', () => {
    expect(wrapper.vm.range).toEqual(true)
  })
  it('Renders props when passed', () => {
    expect(wrapper.vm.progress).toEqual(80)
    expect(wrapper.vm.color).toEqual('success')
    expect(wrapper.vm.imagesrc).toEqual('./images/test.png')
    expect(wrapper.vm.label).toEqual('Test label prop')
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
