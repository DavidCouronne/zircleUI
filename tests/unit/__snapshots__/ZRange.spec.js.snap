// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-range.vue Has the expected html structure 1`] = `
<div
  class="zui disc success range"
  style="width: 30px; height: 30px; margin: -15px 0px 0px -15px; transform: translate3d(0px, 0px, 0px);"
  title="z-range"
  type="range"
>
  <div
    class="z-pulse"
  />
   
  <section
    type="range-bar"
  >
    <svg
      class="scroll"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        r="51"
        style="transform: rotate(0deg); stroke-dasharray: 314.1592653589793; stroke-dashoffset: 62.83185307179587; stroke-width: 9;"
      />
    </svg>
     
    <svg
      class="scroll2"
      style="transform: translate3d(3.7082039324993668px, -11.412678195541844px, 0px);"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="handlebar"
        cx="20"
        cy="20"
        r="8"
      />
    </svg>
  </section>
   
  <section
    class="label"
  >
    
      Test label prop
      
    <!---->
    
      
    
  </section>
   
  <div
    class="z-content"
  >
    <img
      height="100%"
      src="./images/test.png"
      width="100%"
    />
     
    <!---->
  </div>
   
  <div
    class="z-content"
  >
    <span
      class="overflow"
    >
      <p>
         lorem text 
      </p>
    </span>
  </div>
   
  <div />
</div>
`;
